core/src/main/java/com/sequenceiq/cloudbreak/service/altus/AltusMachineUserService.java

```diff
@@ -176,14 +178,19 @@ public DataBusCredential getOrCreateDataBusCredentialIfNeeded(Long stackId) thro
     }
 
     public CdpAccessKeyType getCdpAccessKeyType(StackDto stackDto) {
-        if (!entitlementService.isECDSABasedAccessKeyEnabled(stackDto.getAccountId())) {
-            return CdpAccessKeyType.ED25519;
-        }
-        Optional<Image> image = componentConfigProviderService.findImage(stackDto.getId());
-        if (image.isEmpty()) {
-            return CdpAccessKeyType.ED25519;
+        if (AwsConstants.AwsVariant.AWS_NATIVE_GOV_VARIANT.variant().value().equals(stackDto.getPlatformVariant())) {
+            Optional<Image> image = componentConfigProviderService.findImage(stackDto.getId());
+            if (image.isEmpty()) {
+                throw new CloudbreakRuntimeException("We can't figure out package versions for image because the image is empty.");
+            } else {
+                if (telemetryFeatureService.isECDSAAccessKeyTypeSupported(image.get().getPackageVersions())) {
```

Comment:
a bit strange this info comes from `telemetryFeatureService`. Is this used only for telemetry?

Reply:
no, but there are some restriction where newer key type can only be used from upwards certain versions.

Reply:
thinking about it this comes down to an image validation problem, that we should only allow images from a certain on gov environments.